R/calc_truth.R

```diff
@@ -0,0 +1,342 @@
+#' Calculate "truth" for a species using {ecosystemdata} outputs
+#' 
+#' @description The truth from an ecosystem model is often in different units
+#' than what is needed to sample from. This function takes an {ecosystemdata}
+#' data set and calculates additional true information such as converting
+#' monthly age-composition information into annual data. No sampling is
+#' performed just mathematical operations.
+#' 
+#' @details The following values are calculated within this function:
+#'   * biomass
+#'   * catch
+#'   * numbers-at-age by dividing biomass-at-age by weight-at-age information
+#'   * weight
+#' 
+#' @param data A tibble containing ecosystem model (e.g., Ecopath with Ecosim) 
+#' outputs from [ecosystemdata::load_model()].
+#' @param species_name A string specifying the species name to subset.
+#' 
+#' @return
+#' A nested tibble where each row holds an output type (e.g., biomass, catch, 
+#' numbers, or weight) for a given time step category (e.g., monthly or yearly).
+#' Thus, data within a row in returned output can be expanded to reveal the
+#' full data, e.g., all months for an output type.
+#' 
+#' @examples
+#' data(ewe_nwatlantic_base, package = "ecosystemdata")
+#' data <- ewe_nwatlantic_base
+#' truth <- calc_truth(
+#'   data,
+#'   species_name = "menhaden"
+#' )
+#' 
+#' # To unnest tibble and view the data
+#' truth_long <- tidyr::unnest(truth)
+#'   
+#' # To calculate truth for a vector of species using {purrr}
+#' species_names <- c("menhaden", "striped bass", "bluefish")
+#' truth <- purrr::map_dfr(
+#'   species_names,
+#'   function(x) calc_truth(data, species_name = x)
+#' )
+#' @export
+#' @keywords calc_truth
+calc_truth <- function(data, species_name){
+
+  # Check if the species_name is in the data
+  unique_species <- unique(data[["species"]])
+  if (!species_name %in% unique_species) {
+    cli::cli_abort(c(
+      "x" = "Species {.val {species_name}} not found in the data.",
```

Comment:
```suggestion
      "x" = "Species {.val {species_name}} not found in the {.par data[['species']]}.",
```
I am not actually sure if the suggestion is technically correct but I think that it would be good for the user to be reminded which column of data is being used.

Reply:
Done. Thanks for the suggestion.